src/components/Articles/ViewArticle.js

```diff
@@ -0,0 +1,158 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import PropTypes from 'prop-types';
+import {
+  fetchSpecificArticle,
+  updateArticle,
+} from '../../actions/articleActions';
+import ReadArticle from './ReadArticle';
+import NewArticleForm from './NewArticleForm';
+
+export class ViewArticle extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      description: '',
+      body: '',
+      isEditing: false,
+    };
+    this.toggleEdit = this.toggleEdit.bind(this);
+  }
+
+  componentDidMount() {
+    const { fetchSpecificArticle } = this.props;
+    const {
+      match: {
+        params: { slug },
+      },
+    } = this.props;
+    fetchSpecificArticle(slug);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { articlePayload } = nextProps;
+    const { article: { title, body, description } } = articlePayload;
+    this.setState({ title });
+    this.setState({ body });
+    this.setState({ description });
+
+    if (nextProps.editArticleSuccess === true) {
+      this.setState({ isEditing: false });
+      const { history } = this.props;
+      history.push('/dashboard');
+    }
+  }
+
+
+  handleChange = event => {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  };
+
+  handleEditorChange = value => {
+    this.setState({ body: value });
+  };
+
+  handleSubmit = event => {
+    event.preventDefault();
+    const { title, description, body } = this.state;
+    const payload = {
+      article: {
+        title,
+        description,
+        body,
+      },
+    };
+
+    const { updateArticle } = this.props;
+    const {
+      match: {
+        params: { slug },
+      },
+    } = this.props;
+    updateArticle(slug, payload);
+  };
+
+  resetForm = () => {
+    this.setState({
+      title: '',
+      description: '',
+      body: '',
+    });
+  };
+
+  toggleEdit() {
+    const { isEditing } = this.state;
+    this.setState({ isEditing: !isEditing });
+  }
+
+  render() {
+    const { title, description, body } = this.state;
+    const { loading } = this.props;
+    const { articlePayload, articlePayload: { article } } = this.props;
+    const { isEditing } = this.state;
```

Comment:
Lines 92 and 95 can be put on the same line,
i.e. `const { title, description, body, isEditing } = this.state;`

Lines 93 and 94 can also be put on the same line.

Reply:
@MuhanguziDavid code refactored as advised